plugins/inputs/intel_dlb/intel_dlb.go

```diff
@@ -0,0 +1,470 @@
+//go:generate ../../../tools/readme_config_includer/generator
+//go:build linux
+// +build linux
+
+package intel_dlb
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal/choice"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type IntelDLB struct {
+	SocketPath       string          `toml:"socket_path"`
+	EventdevCommands []string        `toml:"eventdev_commands"`
+	DLBDeviceIDs     []string        `toml:"dlb_device_types"`
+	Log              telegraf.Logger `toml:"-"`
+
+	connection           net.Conn
+	devicesDir           []string
+	rasReader            rasReader
+	maxInitMessageLength uint32
+	logOnce              map[string]error
+}
+
+// DO NOT REMOVE THE NEXT TWO LINES! This is required to embed the sampleConfig data.
+//
+//go:embed sample.conf
+var sampleConfig string
+
+const (
+	defaultSocketPath      = "/var/run/dpdk/rte/dpdk_telemetry.v2"
+	pluginName             = "intel_dlb"
+	eventdevListCommand    = "/eventdev/dev_list"
+	dlbDeviceIDLocation    = "/sys/devices/*/*/device"
+	aerCorrectableFileName = "aer_dev_correctable"
+	aerFatalFileName       = "aer_dev_fatal"
+	aerNonFatalFileName    = "aer_dev_nonfatal"
+	defaultDLBDevice       = "0x2710"
+)
+
+// SampleConfig returns sample config
+func (d *IntelDLB) SampleConfig() string {
+	return sampleConfig
+}
+
+// Init performs validation of all parameters from configuration.
+func (d *IntelDLB) Init() error {
+	var err error
+
+	if d.SocketPath == "" {
+		d.SocketPath = defaultSocketPath
+		d.Log.Debugf("Using default '%v' path for socket_path", defaultSocketPath)
+	}
+
+	err = checkSocketPath(d.SocketPath)
+	d.logErrorOnce("socketPathDoesNotExist", err)
+
+	if len(d.EventdevCommands) == 0 {
+		eventdevDefaultCommands := []string{"/eventdev/dev_xstats", "/eventdev/port_xstats", "/eventdev/queue_xstats", "/eventdev/queue_links"}
+		d.EventdevCommands = eventdevDefaultCommands
+		d.Log.Debugf("Using default eventdev commands '%v'", eventdevDefaultCommands)
+	}
+
+	if err = validateEventdevCommands(d.EventdevCommands); err != nil {
+		return err
+	}
+
+	if len(d.DLBDeviceIDs) == 0 {
+		d.DLBDeviceIDs = []string{defaultDLBDevice}
+		d.Log.Debugf("Using default DLB Device ID '%v'", defaultDLBDevice)
+	}
+
+	err = d.checkAndAddDLBDevice()
+	if err != nil {
+		return err
+	}
+
+	d.maxInitMessageLength = 1024
```

Comment:
Should this be configurable?

Reply:
No, it is ok.
This value is adjusted to buffer size needed for first read (based on: https://github.com/DPDK/dpdk/blob/v22.07/lib/telemetry/telemetry.c#L352)
Afterwards, it is set to `max_output_len` returned from this first read.